src/power_grid_model/_core/power_grid_meta.py

```diff
@@ -205,3 +205,40 @@ def initialize_array(
         dtype=power_grid_meta_data[data_type][component_type].dtype,
         order="C",
     )
+
+
+def attribute_dtype(
```

Comment:
I recall we agreed to  name it `get_attribute_dtype`?

Reply:
I recall we agreed to name it as `attribute_dtype`

Reply:
Wouldn't naming it per class be more intuitive since it's predicate here?

Reply:
@Jerry-Jinfeng-Guo  As in source_dtype, line_dtype?
Its the same effort. No reason to add a whole lot of imports because of it.

Reply:
@figueroa1395 Do you have a preference?

Reply:
I prefer just to have one function with three arguments, as it is already.

Regarding the naming: Since it is intended to be used via the keyword argument `dtype`, I think `attribute_dtype` is fine, or else it is a bit long. If the usage was to be other, then I would prefer the use of the `get_*` prefix, but it is not the case.